[![](helix1.gif)](helix1.gif) |

* * *

####  CS 439 Computational Biology Course Web Site, Fall 1996

* * *

**Professor:** [Steven Salzberg](home.html), salzberg@cs.jhu.edu  
**Time:** Tuesdays and Thursdays, 4:30-5:45 p.m.  
**Location:** Shaffer Hall Room 304

#### This course is co-listed in the Medical School, Department of Biomedical
Information Sciences, as ME 600.802.  
  
---|---  
Textbook: _Introduction to Computational Biology: Maps, Sequences, and
Genomes_ by [Michael Waterman](http://www-hto.usc.edu/people/Waterman.html).
Publisher: Chapman & Hall. Publication date: June 1995. 430 pages. ISBN
0-412-99391-0.

[Homework assignments](assignments.html)

###  Syllabus

**Week 0, Sept. 5.** First day of class. Introduction to the course and text.
Overview of computational biology.

**Week 1, Sept. 10 and 12**. Introduction, biological background. Topic:
Molecular biology for non-biologists. Restriction mapping and interval graphs.

Reading:

  * [ Primer on Molecular Genetics](http://www.gdb.org/Dan/DOE/intro.html), by Denise Casey from DOE (Department of Energy). 
  * Chapter 1 of Waterman. 
  * [ Challenges in the Human Genome Project](readings/robbins1.ps) by Robert Robbins. _IEEE Engineering in Biology and Medicine_ , March 1992, 25-34. 
  * Click [here](genetic-code.ps) for a Postscript image of the genetic code 

**Week 2, Sept. 17 and 19**. Mapping probes and clones. Sequence-tagged site
mapping and why it is computationally ``hard.'' Simulated annealing applied to
contig assembly. Mapping by fingerprinting.

Reading:

  * Waterman, Chapter 2 and pages 101-118 of Chapter 6 (section 6.1.4 is optional). 
  * F. Alizadeh, R. Karp, D. Weisser, and G. Zweig (1995). Physical mapping of chromosomes using unique probes, _J. Computational Biology_ 2:2, 159-184. 
  * Waterman, Sections 4.4.1 and 6.4. (The Alizadeh et al. article is a more thorough treatment of the material in 6.4.) 
  * (Optional) S. Kirkpatrick, C. Gelatt, and M. Vecchi. Optimization by Simulated Annealing. _Science_ 220:4598, May 13, 1983, 671-680. 
**Week 3, Sept. 24 and 27.** No class on Thursday, Sept 26. Instead, there is
a special seminar on Friday.

* * *

**SPECIAL SEMINAR** on Friday, Sept. 27, the first in our Computational
Biology Seminar Series. **[David
Searls](http://cbil.humgen.upenn.edu/~dsearls/home.html)** of SmithKline
Beecham Pharmaceuticals and the University of Pennsylvania will speak on "The
Linguistics of DNA." The lecture is at 4:30pm in Shaffer Hall room 3.

* * *

**Week 4, Oct. 1 and 3.** Sequence alignment, part 1: global distance
alignment, global similarity alignment, and local alignment.

Reading:

  * Waterman, Chapter 9, Sections 9.3 through 9.8. 
  * Temple Smith and Mike Waterman (1981). Identification of Common Molecular Subsequences. _J. Mol. Biol._ 147, 195-197. 
**Week 5, Oct. 8 and 10.** Large-scale shotgun assembly (guest lecture by
Prof. Art Delcher). Sequence alignment, part 2: heuristic alignment methods
including BLAST.

Reading:

  * S. Altschul, W. Gish, W. Miller, E. Myers, and D. Lipman (1990). Basic Local Alignment Search Tool. _J. Mol. Biology_ , 215, 403-410. Some documentation for [BLAST](http://www.ncbi.nlm.nih.gov/BLAST/blast_help.html) is available on-line. 
  * Waterman, Chapter 8. 
  * (Optional) [A survey of sequence comparison algorithms](readings/myers91.ps) (350K) is available from Gene Myers' home page; I've put a copy here for faster downloading. 
**Week 6, Oct. 15 and 17.**

Thursday: **guest lecture by Dr. Granger Sutton of[The Institute for Genomic
Research](http://www.tigr.org)** on whole-genome sequence assembly.

Tuesday: Sequencing by hybridization.

Reading:

  * Granger Sutton, Owen White, Mark Adams, and Anthony Kerlavage. TIGR Assembler: A New Tool for Assembling Large Shotgun Sequencing Projects. _Genomics_ 1:1 (1995), 9-19. An on-line version of this paper is available [here](readings/asmg.paper.ps), courtesy of Granger Sutton. 
  * Waterman, Section 7.2 of Chapter 7. 
  * (Optional) Ramana Idury and Michael Waterman (1995). [A New Algorithm for DNA Sequence Assembly (320K)](readings/idury-waterman.ps) _J. of Computational Biology_ , Vol. 2, 291-306. 
**Week 7, Oct. 23 and 24.** This week we have two special seminars. There will
be no class on Tuesday, Oct. 22. Instead, we have a special seminar on
Wednesday, Oct. 23. We also have a special seminar on Thursday, Oct. 24.

* * *

**SPECIAL SEMINAR** on Wednesday, Oct. 23, the second in our Computational
Biology Seminar Series. **Temple Smith** of Boston University will speak on "A
Review of the Threading Approach to Inverse Protein Structure Prediction." The
lecture is at 4:30pm in Shaffer Hall room 303.

* * *

**SPECIAL SEMINAR** on Thursday, Oct. 24, the third in our Computational
Biology Seminar Series. **Alejandro Schaffer** of the National Center for
Biotechnology Information at the National Institutes of Health will speak on
"Parallel Genetic Linkage Analysis." The lecture is at 4:30pm in Shaffer Hall
Room 3.

* * *

**Week 8, Oct. 29 and 31.** Algorithms for constructing amino acid
substitution matrices. Introduction to Hidden Markov Models for sequence
analysis.

Reading:

  * D. Jones, W. Taylor, and J. Thornton (1992). The rapid generation of mutation data matrices from protein sequences. _Computer Applications in the Biosciences_ (CABIOS) 8:3, 275-282. 
  * Waterman, Chapter 10, especially section 10.5 
  * Kai-Fu Lee, _Automatic Speech Recognition: The Development of the SPHINX System_ , Chapter 1 (on HMMs). Copies of this chapter will be made available in class. 
  * (optional) Anders Krogh, Michael Brown, I. Saira Mian, Kimmen Sjolander, and David Haussler (1993). [ Hidden Markov Models in Computational Biology: Applications to Protein Modeling](readings/hmm.part1.ps) (344K). The figures are available in a separate file, which you get by clicking [ here](readings/hmm.part2.ps) (1,009K). This is technical report UCSC-CRL-93-32, UC Santa Cruz, Dept. of Computer and Information Sciences. A shorter version appeared in _J. Mol. Biology_ , 1993. 
**Week 9, Nov. 5 and 7.** Genetic sequence and map databases and the Human
Genome Project. This topic will be presented in a set of guest lectures by
Prof. Stan Letovsky of the Division of Biomedical Information Sciences.

* * *

**SPECIAL SEMINAR** on Thursday, Nov. 7, the fourth in our Computational
Biology Seminar Series. **Christian Overton** of the University of
Pennsylvania will speak on "High-Throughput Annotation of Genomic Sequences."
The lecture is at 4:30pm in Shaffer Hall Room 3.

* * *

Reading:

  * D.A. Benson, M. Boguski, D.J. Lipman, and J. Ostell (1994). GenBank. _Nucleic Acids Research_ 22(17): 3441-3444. 
  * (Optional) Kenneth Fasman, A.J. Cuticchia, and David Kingsbury (1994). The GDB Human Genome Data Base anno 1994. _Nucleic Acids Research_ 22(17): 3462-3469. 
  * (Optional) Kenneth Fasman, Stanley Letovsky, Robert Cottingham, and David Kingsbury (1995). [Improvements to the GDB Human Genome Data Base](http://gdbdoc.gdb.org/~ken/nar1995.htm). (This is a preprint of a 1995 Nucleic Acids Research article about GDB.) 
**Week 10, Nov. 12 and 14.** The Genome Data Base. (Continuation of Prof.
Letovsky's lectures.)

* * *

**SPECIAL SEMINAR** on Thursday, Nov. 14, the fifth in our Computational
Biology Seminar Series. [Eugene Myers](http://www.cs.arizona.edu/people/gene/)
of the University of Arizona will speak on "Approaches to Whole Genome DNA
Sequencing." The lecture is at 4:30pm in Shaffer Hall room 3.

* * *

**Week 11, Nov. 19 and 21.** On November 19 class will be replaced by a
seminar in the Center for Language and Speech Processing. The seminar is at
4:30pm in Shaffer Hall room 100, by Prof. Alan Biermann of Duke University.
The topic is "Goal-Oriented Multimedia Dialogue with Variable Initiative."

Nov 21: Final lecture by Prof. Letovsky, on computational models of
morphogenesis.

Reading (for next week):

  * Jim Fickett and C.-S. Tung (1992), Assessment of protein coding measures. _Nucleic Acids Research_ 20:24, 6441-6450. 
  * Steven Salzberg (1995). [ Locating Protein Coding Regions in Human DNA using a Decision Tree Algorithm.](exons.ps.gz) _Journal of Computational Biology_ , 2:3 (1995) 473--485.   

  * (Optional, but fun) F.M. Richards (1991). The Protein Folding Problem. _Scientific American_ , January 1991, p. 54-63. 
  * (Optional--more details on decision trees) [A System for Induction of Oblique Decision Trees](jair94.ps) by S.K. Murthy, S. Kasif, and S. Salzberg. _Journal of Artificial Intelligence Research_ 2:1 (1994), 1-32. 
**Week 12, Nov. 26.** Guest lecture by **Dr. Owen White** of [The Institute
for Genomic Research](http://www.tigr.org)** on comparative genomics.

Reading:

  * Sections 14.1 and 14.2 of Waterman. 
**Week 13, Dec. 3 and 5.** Biological background on the structure of genes:
exons, introns, and inter-genic DNA. Description of the gene finding problem.
Computer systems for gene finding. Time permitting: DNA computing; molecular
evolution.

Reading:

  * Leonard Adleman (1994). Molecular Computation of Solutions to Combinatorial Problems. _Science_ , 266, November 11, 1994, 1021-1024. 
  * [A Thousand Billion Billion Sums](readings/comp-dna.ps), reprinted from _The Economist_ , article about Adleman's experiment. 
  * (optional) Rajgopal Srinivasan and George Rose (1995). LINUS: A Hierarchic Procedure to Predict the Fold of a Protein. _PROTEINS: Structure, Function, and Genetics_ **22** , 81-99.   

  * (Optional) [ The Mathematical Basis of Molecular Phylogenetics](http://merlin.mbcr.bcm.tmc.edu:8001/bcdusa/Curric/welcome.html), Chapter 4 of the BioComputing Hypertext Coursebook by Andreas Dress. (This is optional, but you should take a look at least to see what this on-line textbook contains.) 

###  Assignments and grading

The grade will be based on problem sets, programming assignments, and either a
final exam or final project. There are a total of six
[assigments](assignments.html) which count for a total of 70% of the grade.
The final exam accounts for the remaining 30%. With approval of the
instructor, a final project may take the place of the final exam.

###  Useful resources for the course

[Last year's (1995) syllabus](cs439_1995.html)  
[ Home Page for Computational Biology at Hopkins
](http://www.cs.jhu.edu/labs/compbio/home.html)  
[ Home Page for Computer Science at Hopkins ](http://www.cs.jhu.edu/)

