#  Introduction to Bioinformatics

  
PHCL 7611 / BIOM 6682  (2 credits) |

Prof. Larry Hunter  
  
---|---  
Introduction to Bioinformatics |

SoM #2817b  
  
CPH Auditorium (except 9/13 and 11/1 in Sabin Auditorium) |

315-1094  
  
Thursdays, 3:00-4:30pm |

[Larry.Hunter@uchsc.edu](mailto:Larry.Hunter@uchsc.edu)  
  
Office hours by appointment

|

[http://compbio.uchsc.edu/hunter](http://compbio.uchsc.edu/Hunter)  
  
* * *

Course Description | Syllabus |  Grading |  Course Materials  
---|---|---|---  
  
* * *

##  Course description

A gentle introduction to the theory and practice of bioinformatics and
computational biology. Topics include: the analysis of macromolecular
sequences, structures, gene expression arrays and management of the biological
literature.

**Goals for the course:** The course will familiarize students with the tools
and principles of contemporary bioinformatics. By the end of the course,
students will have a working knowledge of a variety of publicly available data
and computational tools important in bioinformatics, and a grasp of the
underlying principles that is adequate for them to evaluate and use novel
techniques as they arise in the future.

**Course Requirements:**

  * **There are no formal prerequisites for this course.** A familiarity with the concepts of molecular biology and comfort using computers and the internet will be assumed. 
  * **Textbook:** _Bioinformatics: Sequence and Genome Analysis,_ by David Mount.     The book is available in the bookstore. 
  * Access to a computer with an internet connection 

###  Syllabus

  1. **What is bioinformatics, and why study it?**
    1. A whirlwind tour of some major bioinformatics resources on the web
    2. An introduction to some computer science and its relationship to bioinformatics
    3. How this class works  

  2. **Macromolecular sequence analysis**
    1. The types and origins of macromolecular sequences
    2. The value of aligning sequences to one another
    3. Dynamic programming, and other approaches to aligning pairs of sequences
    4. Multiple sequence alignment
    5. Hidden Markov models of sequence families
    6. Searching sequence databases
    7. Machine learning and the prediction of sequence characteristics  

  3. **Macromolecular structure analysis**
    1. The types and origins of macromolecular structures
    2. Structure prediction problems
    3. Methods for RNA structure prediction
    4. Methods for Protein secondary structure prediction
    5. Methods for Protein tertiary structure prediction
    6. Molecular Dynamics
  4. **Phylogentic analysis**
  5. **Gene expression array analysis**
  6. **Biological pathway analysis**
  7. **Managing and analyzing the biological literature**

##

* * *

Grading

**Problem sets:** There will be four problem sets, each of which is expected
to take you several hours of work.    These problem sets are intended to give
you hands on experience with the work of computational biology, and should be
learning experiences as well as evaluation tools. The bulk of these problem
sets will involve going to various web sites and doing things.

**Final/Project** :  There will be a **take home final exam handed out at the
end of the last class on November 15, and due by 5pm Friday, November 16** .
If you prefer, you may substitute a course project for the final exam.  If you
want to do a project, you will have to submit a proposal **by October 18**.
Please see me early in the course if you are interested in doing a project
instead of the exam.

### Problem sets

Problem sets and solutions will be posted here.

[Problem set 1](ps1.html)

[Problem set 2](ps2.html)

###  Final

The [final exam](final.html) is due November 19 at 9am

##

* * *

Course materials

###  Schedule updates

Please check here for changes from the normal schedule.

_

  * There will be no class on **September 13**. 
  * Class meets in Sabin Auditorium on **November 1**. _

###  Lecture Notes

Lecture notes will be posted here, generally just before class time. The HTML
versions are best for viewing online, and the PDF versions are better for
printing (but the files are large, and may take a while to download). To read
the PDF files, you may need to download the [Adobe Acrobat PDF file
reader](http://www.adobe.com/products/acrobat/readstep2.html)

HTML versions: [8/30/2001](lecture1)     [9/6/2001](lecture2)
[9/20/2001](lecture3)     [9/27/2001](lecture4)     [10/4/2001](lecture5)
[10/11/2001](lecture6)     [10/18/2001](lecture7)     [10/25/2001](lecture8)
[11/1/2001](lecture9)     [11/8/2001](lecture10)  

PDF versions: [8/30/2001](Lecture1.pdf)     [9/6/2001](Lecture2.pdf)
[9/20/2001](Lecture3.pdf)     [9/27/2001](Lecture4.pdf)
[10/4/2001](Lecture5.pdf)     [10/11/2001](Lecture6.pdf)
[10/18/2001](Lecture7.pdf)     [10/25/2001](Lecture8.pdf)
[11/1/2001](Lecture9.pdf)     [11/8/2001](Lecture10.pdf)  

###  Web sites

This section will grow as the course progresses.  Here are pointers to some of
the web sites I discuss in class.  


* **Local Resources**

  * [Center for Computational Pharmacology](http://compbio.uchsc.edu)
  * [ CU Denver/UCHSC Center for Computational Biology](http://www.cudenver.edu/ccb/) 

* **Listings of bioinformatics web sites**

  * [ Nucleic Acids Research Database Issue 2001](http://nar.oupjournals.org/content/vol29/issue1/#ARTICLES) (with [Database of Databases](http://www3.oup.co.uk/nar/db2001/c/) )
  * [ Weizmann Institute list of bioinformatics web sites](http://bioinformatics.weizmann.ac.il/mb/molecular_biol_databases.html)
  * [Kyoto list of bioinformatics web sites](http://www.genome.ad.jp/kegg/kegg4.html)
  * [ Google bioinformatics entry](http://directory.google.com/Top/Science/Biology/Bioinformatics/)

* **Bioinformatics journals, conferences and society**

  * [International Society for Computational Biology](http://www.iscb.org)
  * [Bioinformatics](http://bioinformatics.oupjournals.org/) (journal)
  * [Journal of Computational Biology](http://www.cs.sandia.gov/jcb/)
  * [Pacific Symposium on Biocomputing](http://psb.stanford.edu) (Free full text proceedings)
  * [Intelligent Systems for Molecular Biology (ISMB)](http://ismb01.cbs.dtu.dk)
  * [RECOMB](http://recomb2001.gmd.de/)

* **Journalistic coverage of bioinformatics** (useful for keeping up with commercial developments)

  * [GenomeWeb.com](http://www.genomeweb.com/) (companion to Genome Technology, free magazine)
  * [Bioinform](http://www.bioinform.com/)

* **[NCBI**](http://www.ncbi.nlm.nih.gov/) National Center for Biotechnology Information

  * [Entrez](http://www.ncbi.nlm.nih.gov/Entrez/)
  * [BLAST](http://www.ncbi.nlm.nih.gov/BLAST/)
  * [Education pages](http://www.ncbi.nlm.nih.gov/Education)

* [**PDB**](http://www.rcsb.org/pdb/) Protein DataBank, a database of macromolecular structures 

### Background

For those of you who are taking the course without a background in both
molecular biology _and_ computer science, here are background materials that
will help you come up to speed in the areas you are less familiar with.

####  Computer Science Background

I believe that to be successful as a bioinformatician, you need to be able to
design and create complex software systems. There is no substitute for
experience, so the best training is to write a reasonably large piece of
software under the close supervision of an expert. There's nothing like
experience to drive home the value of appropriate data structures, correct
algorithm choice, clear interfaces and good documentation. In terms of topics,
I'd suggest at least (1) data structures, (2) algorithms and (3) software
engineering. Here are some introductory books and other resources to get you
started down that path:  


  * _Computer Science, an Overview_ J. Glenn Brookshear, 6th edition, 1999\. An excellent introductory textbook, which you can [ purchase at amazon](http://www.amazon.com/exec/obidos/ASIN/020135747X) , or you might want to look for used copies on the net.
  


  * _[The Structure and Interpretation of Computer Programs](http://sicp.arsdigita.org/text/sicp) _ Hal Abelson and Gerry Sussman. The best textbook on software engineering. Not a gentle introduction, but a great one. The book is now free online. In addition, there is a [ complete and free set of video lectures by Abelson and Sussman themselves](http://64.64.103.140/ind.html#1986) available from ArsDigita university. These lectures were given in 1986, but they are still an excellent introduction to the principles and practices of software engineering. The entire online university is worth exploring: <http://aduni.org>
  


  * Also, a lot of computer science is knowing the best algorithms developed for a wide variety of particular tasks. My favorite reference book for algorithms (and a good investment if you are serious about computer science) is Cormen, Leiserson & Rivest, _Introduction to Algorithms_ McGraw-Hill, 1990 (yes, that's OK) You can [ purchase it at Amazon](http://www.amazon.com/exec/obidos/ASIN/0262031418) or look for a used copy.
  
  

####  Molecular Biology Background

Being a successful computational biologist requires more knowledge of the
domain than most applied computer science. Decent biological intuition is
crucial to avoid making painful design errors. Biology is taught in a spiral,
where one keeps returning to the same topics, each time with increasing levels
of sophistication and subtlety. These links will take you around the lowest
level of the spiral once:  


  * [ An Introduction to Molecular Biology for the Computer Scientist](http://www.aaai.org//Library/Books/Hunter/01-Hunter.pdf) Ch.1 of Hunter, L. _[Artificial Intelligence and Molecular Biology](http://www.aaai.org//Library/Books/Hunter/hunter.html) _ MIT Press 1993
  


  * [The MIT Biology Hypertextbook](http://esg-www.mit.edu:8001/esgbio/chapters.html)

